enhancements/cloud-integration/openstack/cloud-provider-config-upgrade.md

```diff
@@ -0,0 +1,327 @@
+---
+title: OpenStack Cloud Provider Config Upgrade
+authors:
+  - "@stephenfin"
+  - "@mdbooth"
+reviewers:
+  - TBD
+approvers:
+  - "@jspeed"
+  - TBD
+api-approvers: # in case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers)
+creation-date: 2022-01-18
+last-updated: 2022-01-25
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OSASINFRA-2758
+see-also:
+replaces:
+superseded-by:
+---
+
+# OpenStack Cloud Provider Config Upgrade
+
+## Summary
+
+Cloud configuration is stored in the user-managed ConfigMap
+`cloud-provider-config` in `openshift-config`.  In 4.11 we want to switch
+OpenStack clusters from the legacy cloud provider to the external Cloud
+Controller Manager (CCM). The configuration for the legacy and external cloud
+providers are similar, but not identical.
+
+This enhancement will describe how we can ensure correct configuration is passed
+to the cloud provider in use during and after an upgrade, and how we can prevent
+an upgrade to 4.11 if the configuration cannot be upgraded.
+
+## Motivation
+
+The legacy cloud provider is dead. Long live the external cloud provider.
+
+### Goals
+
+- No user action is required in the common case
+- Upgrade legacy cloud provider to external cloud controller manager where possible
+- Mark cluster as non-upgradable where upgrade to external CCM is not possible
+
+### Non-Goals
+
+- Modify user-managed configuration
+- Manage cloud providers for other, non-OpenStack platforms
+
+## Proposal
+
+- The change will be implemented in CCO
```

Comment:
Nit: CCO could be one of multiple different operators within OpenShift, I'd recommend expanding the acronym on the first usage in case people think you're talking about implementing changes in the cloud-credential-operator here